### PR TITLE
Short titles

### PR DESCRIPTION
**Pull request by simpsonw**
Add the ability to give exhibit pages an alternative title that displays only in the navigation menu.  For example, a page could have a full title of "Late Medieval Catholicism: Lay Piety and Clerical Theology" that's shortened to "Late Medieval Catholicism" in the menu.  This functionality was described in issue #5.

### CODE REVIEW

**Comment by zerocrates** [comment]
This probably should alter the other nav-related functions/helpers as well: the older "stacked" nav, the previous/next links, the "Up" links... pretty much [this area](https://github.com/omeka/plugin-ExhibitBuilder/blob/master/helpers/ExhibitPageFunctions.php#L70-L261). I could be wrong about the expectation but it does seem appropriate to do this in all those nav-ish places, particularly as the presence of the feature will encourage longer "normal" titles.

There's enough places that use the title for nav-ish stuff that it probably makes sense to implement a dynamic property (using [getProperty](https://github.com/omeka/Omeka/blob/master/application/models/Item.php#L178)) so it can be a simple swap-out of `title` for `nav_title` in `metadata()` calls, where `nav_title` handles the short/regular title fallback you've already got in your single location here.

**Comment by zerocrates** [comment]
Beyond that, this looks pretty good to me.

**Comment by simpsonw** [comment]
@zerocrates I think I got all the areas you mentioned in ExhibitPageFunctions.php.  I renamed the property to `short_title` (both in the code and the db) since that more clearly describes what it is and added a dynamic property called `menu_title` that returns `short_title` if it's set and not empty and the `title` attribute otherwise.  Please let me know if there's anything else I can do!

File: models/ExhibitPage.php [hunk 1]
```diff
@@ -405,4 +412,24 @@ public function setFixChildrenOnDelete($fix)
...
+     * Get a property for display.
+     *
+     * @param string $property
+     * @return mixed
+     */
+    public function getProperty($property)
+    {
+        switch($property) {
+            case 'menu_title':
+                if(isset($this->short_title) && !empty($this->short_title)) {
```
Comment (zerocrates) [inline models/ExhibitPage.php]: `!empty` should be sufficient by itself here, as it provides the same error-suppressing properties that `isset` does.

Comment (simpsonw) [inline models/ExhibitPage.php]: Good point!  I've removed the call to `isset()`.

**Comment by zerocrates** [comment]
@mebrett I guess let's just continue to use the PR issue, but I've pulled the feature into master

**Comment by mebrett** [comment]
Sounds good, thank you.

**Comment by mebrett** [comment]
It works fine except for one issue in Seasons.

If the Exhibit Builder page navigation style for Seasons is set as full-width (under the theme config) then none of the navigation text displays at all. The space is there, but no text.

With short titles: 
screen shot 2017-07-26 at 4 17 30 pm

Older version of Exhibit Builder:
screen shot 2017-07-26 at 4 19 38 pm

**Comment by zerocrates** [comment]
Okay yeah, I missed a couple little errors in the PR: in a few places it tries to access `menu_title` directly as a property, but it needs to be using `metadata()` instead.

File: helpers/ExhibitPageFunctions.php [hunk 2]
```diff
@@ -128,7 +128,7 @@ function exhibit_builder_child_page_nav($exhibitPage = null)
     $children = exhibit_builder_child_pages($exhibitPage);
     $html = '<ul class="exhibit-child-nav navigation">' . "\n";
     foreach ($children as $child) {
-        $html .= '<li><a href="' . html_escape(exhibit_builder_exhibit_uri($exhibit, $child)) . '">' . html_escape($child->title) . '</a></li>';
+        $html .= '<li><a href="' . html_escape(exhibit_builder_exhibit_uri($exhibit, $child)) . '">' . html_escape($child->menu_title) . '</a></li>';
```
Comment (zerocrates) [inline helpers/ExhibitPageFunctions.php]: The `html_escape($child->menu_title)` here should be `metadata($child, 'menu_title')`

File: helpers/ExhibitPageFunctions.php [hunk 3]
```diff
@@ -92,18 +92,18 @@ function exhibit_builder_page_nav($exhibitPage = null)
...
 
         $html .= "<li>\n<ul class=\"exhibit-nav-level-$levelNumber\">\n";
         $levelNumber +=1;
-        
+
         foreach ($pageSiblings as $pageSibling) {
             $html .= '<li' . ($pageSibling->id == $page->id ? ' class="current"' : '') . '>';
             $html .= '<a class="exhibit-page-title" href="' . html_escape(exhibit_builder_exhibit_uri($exhibit, $pageSibling)) . '">';
-            $html .= html_escape($pageSibling->title) . "</a></li>\n";
+            $html .= html_escape($pageSibling->menu_title) . "</a></li>\n";
```
Comment (zerocrates) [inline helpers/ExhibitPageFunctions.php]: should be `metadata($pageSibling, 'menu_title')`

File: helpers/ExhibitPageFunctions.php [hunk 4]
```diff
@@ -92,18 +92,18 @@ function exhibit_builder_page_nav($exhibitPage = null)
     $levelNumber = 1;
     
     foreach ($pagesTrail as $page) {
-        $linkText = $page->title;
+        $linkText = $page->menu_title;
```
Comment (zerocrates) [inline helpers/ExhibitPageFunctions.php]: This is a problem but it looks like `$linkText` is totally unused anyway so it can just be deleted.

**Comment by zerocrates** [comment]
@mebrett OK, I believe that should be fixed

**Comment by mebrett** [comment]
@zerocrates Everything looks as it should now, thanks.

**Pull request merged**